core/src/main/java/me/lokka30/treasury/plugin/core/command/subcommand/migrate/MigrateSubcommand.java

```diff
@@ -194,70 +185,11 @@ private void sendMigrationMessage(@NotNull CommandSource sender, @NotNull Migrat
-                        ),
                         placeholder("non-migrated-currencies", Utils.formatListMessage(migration.nonMigratedCurrencies()))
                 )
         );
     }
 
-    private Phaser establishCurrencies(@NotNull MigrationData migration) {
-        CompletableFuture<Collection<UUID>> fromCurrencyIdsFuture = new CompletableFuture<>();
-
-        // Initialize phaser with a single party: currency mapping completion.
-        Phaser phaser = new Phaser(1);
-
-        migration.from().provide().retrieveCurrencyIds(new PhasedFutureSubscriber<>(phaser, fromCurrencyIdsFuture));
-
-        fromCurrencyIdsFuture.thenAccept(fromCurrencyIds -> {
-            for (UUID fromCurrencyId : fromCurrencyIds) {
-
-                // Fetch from currency.
-                CompletableFuture<Currency> fromCurrencyFuture = new CompletableFuture<>();
-                migration.from().provide().retrieveCurrency(
-                        fromCurrencyId,
-                        new PhasedFutureSubscriber<>(phaser, fromCurrencyFuture)
-                );
-                fromCurrencyFuture.whenComplete(((currency, throwable) -> {
-                    if (throwable != null) {
-                        migration.debug(() -> "Unable to locate reported currency with ID '&b" + fromCurrencyId + "&7'.");
-                    }
-                }));
-
-                fromCurrencyFuture.thenAccept(fromCurrency -> {
-
-                    // Fetch to currency.
-                    CompletableFuture<Currency> toCurrencyFuture = new CompletableFuture<>();
-                    migration.to().provide().retrieveCurrency(
-                            fromCurrencyId,
-                            new PhasedFutureSubscriber<>(phaser, toCurrencyFuture)
-                    );
-                    toCurrencyFuture.whenComplete(((toCurrency, throwable) -> {
-                        if (toCurrency == null) {
-                            // Currency not found.
-                            migration.nonMigratedCurrencies().add(fromCurrency.getPrimaryCurrencyName());
-                            migration.debug(() -> "Currency of ID '&b" + fromCurrency.getPrimaryCurrencyName() + "&7' will not be migrated.");
-                        } else {
-                            // Currency located, map.
-                            migration.migratedCurrencies().put(fromCurrency, toCurrency);
-                            migration.debug(() -> "Currency of ID '&b" + fromCurrency.getPrimaryCurrencyName() + "&7' will be migrated.");
-                        }
-                    }));
-                });
-            }
-            phaser.arriveAndDeregister();
-        });
-
-        return phaser;
-    }
-
     private <T extends Account> Phaser migrateAccounts(@NotNull MigrationData migration, @NotNull AccountMigrator<T> migrator) {
         // Initialize phaser with a single party: migration completion.
         Phaser phaser = new Phaser(1);
```

Comment:
what happened to all of the deleted content? has it been moved?

Reply:
it's not needed anymore.

Reply:
@Jikoo are you okay with this?

Reply:
Looks good to me, no reason to keep around currency migration/translation code when currencies are global. 👍